jerry-core/api/jerry.c

```diff
@@ -601,6 +601,34 @@ jerry_get_global_object (void)
   return ecma_make_object_value (global_obj_p);
 } /* jerry_get_global_object */
 
+/**
+ * Check if the specified value is an instance of the given prototype.
+ *
+ * See also ECMA-262 v5.1, 11.8.6
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return true  - if the specified value is an instance of the given prototype
+ *         false - otherwise
+ *         error - if value argument is not an object or prototype argument is not a function.
+ *
+ */
+jerry_value_t
+jerry_value_instanceof (const jerry_value_t value, /**< object value */
+                        const jerry_value_t prototype) /**< prototpye function */
+{
+  jerry_assert_api_available ();
+
+  if (ecma_is_value_error_reference (value))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p)));
+  }
+
+  ecma_object_t *proto_obj_p = ecma_get_object_from_value (prototype);
```

Comment:
Couldn't it cause an assertion fail if this function is called like `jerry_value_instanceof (jerry_create_number (5), jerry_create_number (5))`, due to the second argument is not an object?